### DISCUSSION TITLE
Failed to find instances

### DISCUSSION THREAD

**Discussion by melinasolalvarez-blip** [talmolab/sleap#2687] [Help!] [answered]
Hi! Hope you are all having a good day.
I'm writing here because I'm trying to use SLEAP to automate a behavioral task in rats that is somewhat similar to novel object recognition, but across multiple sessions (for a retrieval-induced forgetting protocol).

During my undergraduate thesis, I previously used SLEAP to track flies in arenas. I managed to get decent tracking results, although I couldn't fully preserve identities consistently before running out of time and I ended up using Tracker to finish my thesis.

Now I just started my PhD and I thought this new setup would actually be easier to track: each rat is always alone in its cage during the task, and each video usually contains two cages recorded in parallel.

My long-term goal is to combine SLEAP tracking with FIJI-based object recognition. Ideally, I would like to estimate when the rat is sniffing the object versus performing other behaviors by overlapping both datasets. I also saw people using another model for the object in SLEAP but since usually the objects change and there are a lot of them I think it's not ideal for my case, but I would appreciate any suggestion on that matter.

I installed the latest version of SLEAP and followed the tutorial to refresh my knowledge, but I'm struggling to get the model to detect any instances at all.

Here’s what I already tried:

Started with parameters similar to the tutorial, adjusting mainly the animal size settings to match the rats.
Training takes quite a long time, but after inference no instances are detected.
I then tried the parameters suggested by the Config Picker tool for my videos, but obtained the same result.
Increased the number of labeled frames progressively (currently at 25 labeled frames).
Still, the model does not detect any instances in any frame.
I ran the sleap doctor but didn't see anything that seems to be the problem to me

At this point I'm a bit lost about what else to try. I suspect the issue could be related to the relatively poor video quality, but I'm not sure.

I'm attaching screenshots of the configurations and an example frame from the videos. Any advice would be greatly appreciated!
Training pipeline:
<img width="831" height="650" alt="image" src="https://github.com/user-attachments/assets/b8c7d3f2-9947-474f-8796-fb46c36e91df" />
<img width="467" height="238" alt="image" src="https://github.com/user-attachments/assets/9bf8b3b3-29fa-4c2f-98ef-d805bdbf8ada" />
Centroid configuration:
<img width="898" height="691" alt="image" src="https://github.com/user-attachments/assets/b9baaa6d-9c18-42ae-ba75-5c400170bf6a" />
with 2 min hard keypoints and none max hard keypoints
Centered instance configuration:
<img width="913" height="699" alt="image" src="https://github.com/user-attachments/assets/51f2303d-1a5b-423a-af75-aa58bf86a5eb" />
Example from the video:
<img width="807" height="585" alt="image" src="https://github.com/user-attachments/assets/121467d6-b1f3-4fef-8f77-e4cb9ebf0ffb" />
<img width="811" height="454" alt="image" src="https://github.com/user-attachments/assets/cc8bf87f-f784-40ae-95dc-54937ba93912" />
Training result 
<img width="478" height="420" alt="image" src="https://github.com/user-attachments/assets/5c703288-19e7-4c23-814b-b16b6d082348" />

EDIT: I noticed even though I installed the program last thursday following the tutorial it installed version .1 instead of .3, I corrected that and I'm currently running training again to see if it solves the problem. I leave the updated doctor results:

SLEAP System Diagnostics
========================
Generated: 2026-05-12 13:04:26

[Platform]
  OS:        Windows 11
  Platform:  Windows-11-10.0.26200-SP0
  Machine:   AMD64
  Processor: Intel64 Family 6 Model 186 Stepping 2, GenuineIntel
  RAM:       12.4 GB / 15.7 GB
  Disk:      420.2 GB / 453.2 GB

[Python]
  Version:     3.13.7
  Executable:  C:\Users\melis\AppData\Roaming\uv\tools\sleap\Scripts\python.exe
  Prefix:      C:\Users\melis\AppData\Roaming\uv\tools\sleap

[UV]
  Version:         uv 0.11.11 (ed7b06001 2026-05-06 x86_64-pc-windows-msvc)
  Path:            C:\Users\melis\.local\bin\uv.EXE
  Cache Dir:       C:\Users\melis\AppData\Local\uv\cache
  Tool Dir:        C:\Users\melis\AppData\Roaming\uv\tools
  Tool Bin Dir:    C:\Users\melis\.local\bin
  Python Dir:      C:\Users\melis\AppData\Roaming\uv\python
  Installed Tools: sleap

[UV Config]
  Default Python:    (not configured)
  Resolved Python:   C:\Users\melis\miniforge3\python.exe
  Python Preference: managed (default)
  Resolution:        highest (default)
  Index Strategy:    first-index (default)
  Prerelease:        if-necessary (default)

[Conda]
  Status:             installed but not activated
  Version:            conda 25.11.0
  auto_activate_base: False

[GPU / CUDA]
  NVIDIA Driver: 581.95 (CUDA 13.0)
  GPU 0:         NVIDIA GeForce RTX 4050 Laptop GPU (5921 MB free / 6141 MB)
  PyTorch:       v2.11.0+cu130 (CUDA 13.0)

[Packages]
  sleap:          v1.6.3 (pip)
  sleap-io:       v0.7.0 (pip)
  sleap-nn:       v0.2.0 (pip)
  numpy:          v2.4.4 (pip)
  h5py:           v3.16.0 (pip)
  PySide6:        v6.11.0 (pip)
  opencv-python:  v4.13.0.92 (pip)
  torch:          v2.11.0+cu130 (pip)
  imageio:        v2.37.3 (pip)
  imageio-ffmpeg: v0.6.0 (pip)
  av:             v15.1.0 (pip)

[CLI Binaries]
  sleap:
    Path:      C:\Users\melis\.local\bin\sleap.EXE
    Real Path: C:\Users\melis\.local\bin\sleap.exe
    Source:    pip
  sleap-nn-track:
    Path:      C:\Users\melis\.local\bin\sleap-nn-track.EXE
    Real Path: C:\Users\melis\.local\bin\sleap-nn-track.exe
    Source:    pip
  ffmpeg (imageio):
    Path:
C:\Users\melis\AppData\Roaming\uv\tools\sleap\Lib\site-packages\imageio_ffmpeg\binaries\ffmpeg-win-x86_64-v7.1.exe
    Source:    imageio-ffmpeg
    Python:    v7.1-essentials_build-www.gyan.dev

[PATH (relevant entries)]
  C:\Users\melis\AppData\Roaming\uv\tools\sleap\Lib\site-packages\cv2\../../x64/vc17/bin
  C:\Users\melis\.local\bin
  C:\Users\melis\AppData\Local\Programs\Python\Python313\Scripts[/]
  C:\Users\melis\AppData\Local\Programs\Python\Python313[/]

Thank you for taking the time to read this

**Comment by gitttt-1234** [comment, marked as answer]
Hi @melinasolalvarez-blip !

Could you try the following?
- For centroid, decrease input scaling to 0.25 (if this doesn't work still, try keeping input scale as 0.5 and increase the max stride to 64)
- For centered instance, set crop size to auto (check-box) and keep input scaling to 1.0

For speeding up training, in the main training pipeline window, set the Data pipeline (under Performance) to cache in disk and increase num workers to 2 (or 4 to make it even faster, but 4 might throw insufficient memory depending on the system).

Let us know if you have any questions!

Thanks,

Divya

**Reply by melinasolalvarez-blip** [reply]
Thank you so much! That worked well :) 
(PS. sorry for the late reply, some personal problems came up and I couldn't try until now)

**Reply by melinasolalvarez-blip** [reply]
I'm having some trouble again. I was able to do the human-in-the-loop training for a while, but then I realized I had accidentally changed some parameters because I used the mouse wheel to scroll down to the Run button, and it can change the configuration.
When I went back and checked, I saw that the last six training runs had been done using the bottom-up configuration instead of top-down.
I already have 200 labeled frames, so I went back to this post, reset the configuration to what we had discussed, and tried training from scratch using those 200 labeled frames. Unfortunately, the same thing happened again.

<img width="594" height="516" alt="sleapinference" src="https://github.com/user-attachments/assets/69a88c69-ef95-4709-9617-0c47a69355b6" />

I'm really sorry to keep bothering you with this issue, but I'm out of ideas about what else to try.

**Reply by gitttt-1234** [reply]
Hey @melinasolalvarez-blip !

That's strange, could you share the configs here and upload the slp file / video [here](slp.sh) so that we could take a closer look?

Thanks,

Divya

**Reply by melinasolalvarez-blip** [reply]
Thank you so much, this are the configurations I tried 
<img width="951" height="628" alt="image" src="https://github.com/user-attachments/assets/b4505377-e6bc-463a-a70b-0de66f2aff08" />
<img width="946" height="272" alt="image" src="https://github.com/user-attachments/assets/87517b3c-ad6e-4db2-83b6-1c9e8eec28ca" />
<img width="942" height="555" alt="image" src="https://github.com/user-attachments/assets/aff07963-38e3-45c9-9005-7bdd0911b617" />
<img width="317" height="225" alt="image" src="https://github.com/user-attachments/assets/026c79c8-f393-4cba-a742-802207933c1f" />
<img width="956" height="635" alt="image" src="https://github.com/user-attachments/assets/e814c665-09b4-4dad-aebd-394784c51be1" />
<img width="358" height="278" alt="image" src="https://github.com/user-attachments/assets/766e4c0f-bd96-446a-9b10-b53950515dc1" />

I'll upload the file there now

**Reply by gitttt-1234** [reply]
Hi @melinasolalvarez-blip !

Thanks for sharing the configs! A couple of things to try:

- Centroid: Filters to 32 and Filters Rate to 1.50. Right now they're at 16 / 2.00, which is less model capacity than the config you started with before.
- Did you experiment with `tailstart1` as the anchor instead of `torso`? In a lot of cases the tail base ends up being a more reliable visual landmark for detection.

If neither of those helps, would you mind uploading your .slp and video to slp.sh? If your video files are large or you have a lot of videos, rather than uploading everything, you can export an embedded package: in the GUI, go to Predict → Export Labels Package… → "Labeled frames". This saves a .pkg.slp with the images embedded, which you can upload to slp.sh instead of the slp project with all the raw videos.

Thanks,

Divya

**Reply by melinasolalvarez-blip** [reply]
Hi! Thank you so much for taking the time to help me, I tried those two things today but the problem unfortunately persists, I'll upload it there
Thanks again
EDIT: this is the link to the slp.sh, just in case, https://slp.sh/aLVqEO/labels.slp

**Reply by gitttt-1234** [reply]
Hey @melinasolalvarez-blip !

Thanks for uploading the data! I trained a topdown model and it seems to be working well for me on your data. Here are the configs I used:

[centroid_config.yaml](https://github.com/user-attachments/files/30720758/centroid_config.yaml)
[centered_instance_config.yaml](https://github.com/user-attachments/files/30720760/centered_instance_config.yaml)


Let us know if these work for you!

Thanks,

Divya
